lab_1_keywords_tfidf/main.py

```diff
@@ -4,106 +4,138 @@
 """
 from typing import Optional, Union
 
+import math
+import string
+
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
     Removes punctuation, casts to lowercase, splits into tokens
-
     Parameters:
     text (str): Original text
-
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(text, str):
+        return None
+    for punc in string.punctuation:
+        text = text.replace(punc, '')
+        clean_words = text.lower().strip().split()
```

Comment:
having this level of identation, you lower, strip and split the text as many times as there are symbols in `strings.punctuation`
it would be enough to do it just once after the cycle is finished

also, could you clarify why you need to use `strip` here

Reply:
just accidentally put the line `clean_words =` under the for loop
and yeah, shouldn't have used strip cause i already have split
fixed!